src/Gateway/TransactionGateway.php

```diff
@@ -73,7 +73,10 @@ public function sendPending(int $ct, $lang = 'en')
      */
     public function cancel(?string $id, ?string $ref, $lang = 'en')
     {
-        $response = $this->request('DELETE', sprintf('/creditor/transaction%s%s%s', isset($id) || isset($ref) ? '?' : null, isset($id) ? "id=$id" : null, isset($ref) ? sprintf("%ref=$ref", isset($id) ? '&' : null) : null, [], $lang));
+        $queryPrefix = isset($id) || isset($ref) ? '?' : null;
+        $queryId = isset($id) ? "id=$id" : null;
```

Comment:
Hey @rvvincelli it's either id or ref, if both are set only the id is taken into account, so I guess we could make the Line 78 a whole lot simpler

Reply:
Hi @koen-serry , thanks for having a look!

To be fair, in the API docs this is not mentioned, see [here](https://www.twikey.com/api/index.html#remove-a-transaction) (they are both optional). I'd rather have this additional complexity you mention of id vs ref be swallowed by the endpoint, rather than the client - one never know if the default behavior might change in the future etc.

But if you really like me to, I can follow your point and adjust the code.

Please let me know, thanks,

Reply:
Hi @rvvincelli thanks for the feedback, I wasn't aware that it wasn't mentioned in the documentation. We'll adapt it right away. Thanks!

Reply:
Hi @koen-serry , graag gedaan! Resolving this one for now (you may re-open it if you feel so). Thanks,